### ISSUE TITLE
Empty nested payload objects set to null

### ISSUE THREAD

**Issue by jeremymailen** [lagoshny/ngx-hateoas-client#116]
Hi @lagoshny, thank you for the well crafted library.

I think I might have uncovered a bug or at least behavior I'm curious to understand.

Here if we are parsing a nested property of the body, we set it to null instead of leaving it alone if it's an empty object or array: https://github.com/lagoshny/ngx-hateoas-client/blob/c24c7f048d71077b285e300fcf2f41457215bc81/projects/ngx-hateoas-client/src/lib/util/resource.utils.ts#L222

For example if someone is patching or updating on object with data like this:

```json
  "body": {
    "id": "675...",
    "env": {},
    "created": "2023-12-..."
  }
```

it's transformed into this in the call:

```json
  "body": {
    "id": "6754...",
    "env": null,
    "created": "2023-12-..."
  }
```

This is problematic if you have nested raw data in your model such as a jsonb or array type and your backend model expects this value to be non-nullable.

Is this transformation intentional or accidental? If appropriate, may I submit a PR to fix?

**Comment by lagoshny** [comment]
Hi @jeremymailen thank you for the pull request.

The lib transforms empty object `{}` to `null` value because on Java side it is problem to parse an empty object `{}` better to consider him as `null` value.
For arrays, yes it is a bug, arrays should not be passed as `null`s when empty.

I will merge your PR and also add new library configuration param  to disable empty objects to `null` transformation to save backward compatibility.

**Comment by lagoshny** [comment]
@jeremymailen published new lib version `3.7.0`. You need to pass new configuration param to disable empty object to `null` convertation:

````ts
    halFormat: {
      json: {
        convertEmptyObjectToNull: false
      }
    }
````
